File: ferubi/Packmaster_Skoiat_Pizak.pl
my $mob1;
my $mob2;
my $mob3;
my $mob4;
my $entid1;
my $entid2;
my $entid3;
my $entid4;
my $mobnpc1;
my $mobnpc2;
my $mobnpc3;
my $mobnpc4;


sub EVENT_AGGRO {
   quest::settimer(1,45);
}

sub EVENT_TIMER {
  if($timer == 1) {
   $entid1 = quest::spawn2(quest::ChooseRandom(284098,284107),0,0,-170,-884,3,114); # NPC(s): #Rav_Pizak (284098), #Rav_Pizak (284107)
   $entid2 = quest::spawn2(quest::ChooseRandom(284098,284107),0,0,-167,-792,3,136); # NPC(s): #Rav_Pizak (284098), #Rav_Pizak (284107)
   $entid3 = quest::spawn2(quest::ChooseRandom(284098,284107),0,0,-170,-884,3,114); # NPC(s): #Rav_Pizak (284098), #Rav_Pizak (284107)
   $entid4 = quest::spawn2(quest::ChooseRandom(284098,284107),0,0,-167,-792,3,136); # NPC(s): #Rav_Pizak (284098), #Rav_Pizak (284107)

   $mob1 = $entity_list->GetMobID($entid1);
   $mob2 = $entity_list->GetMobID($entid2);
   $mob3 = $entity_list->GetMobID($entid3);
   $mob4 = $entity_list->GetMobID($entid4);

   $mobnpc1 = $mob1->CastToNPC();
   $mobnpc2 = $mob2->CastToNPC();
   $mobnpc3 = $mob3->CastToNPC();
   $mobnpc4 = $mob4->CastToNPC();

   $mobnpc1->AddToHateList($npc->GetHateTop()); 
   $mobnpc2->AddToHateList($npc->GetHateTop());
   $mobnpc3->AddToHateList($npc->GetHateTop()); 
   $mobnpc4->AddToHateList($npc->GetHateTop());
  }
}

sub EVENT_DEATH_COMPLETE {
  quest::stoptimer(1);
}

sub EVENT_COMBAT {
  if($combat_state == 0) {
    quest::stoptimer(1);
    quest::depopall(284098);
    quest::depopall(284107);
  }
}

File: ferubi/Packmaster_Skoiat_Pizak.lua
-- NPC(s): #Rav_Pizak (284098), #Rav_Pizak (284107)
guard_list = { 284098, 284107 };

function event_combat(e)
	-- during combat, reaggro the guards on current target every 30 seconds (ensures no idle guards)
	if e.joined then
		aggro_guards(e.self:GetHateTop());
		eq.set_timer('aggro_guards', 30 * 1000);
		eq.set_timer('spawnadds', 45 * 1000);
		eq.stop_timer('depopadds');
	else
		eq.stop_timer('aggro_guards');
		eq.stop_timer('spawnadds');
		eq.set_timer('depopadds', 60 * 1000);
	end
end

function event_timer(e)
	if e.timer == 'aggro_guards' then
		aggro_guards(e.self:GetHateTop());
	elseif e.timer == 'spawnadds' then
		eq.spawn2(eq.ChooseRandom(284098,284107),0,0,-170,-884,3,114):AddToHateList(e.self:GetHateRandom(),1);
		eq.spawn2(eq.ChooseRandom(284098,284107),0,0,-167,-792,3,136):AddToHateList(e.self:GetHateRandom(),1);
		eq.spawn2(eq.ChooseRandom(284098,284107),0,0,-170,-884,3,114):AddToHateList(e.self:GetHateRandom(),1);
		eq.spawn2(eq.ChooseRandom(284098,284107),0,0,-167,-792,3,136):AddToHateList(e.self:GetHateRandom(),1);
	elseif e.timer == 'depopadds' then
		eq.stop_timer('depopadds');
		eq.depop_all(284098);
		eq.depop_all(284107);
	end
end

function aggro_guards(mob)
	for _, guard in pairs(guard_list) do
		local guard_mob = eq.get_entity_list():GetNPCByNPCTypeID(guard);
		if guard_mob ~= nil then
			guard_mob:AddToHateList(mob);
		end
	end
end
